lib/ContractInteract/ProxyFactory.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const Web3 = require('web3');
+const AbiBinProvider = require('./../AbiBinProvider');
+const Contracts = require('../Contracts');
+
+const Utils = require('../../utils/Utils');
+const ContractName = 'ProxyFactory';
+
+/**
+ * Class to interact with ProxyFactory contract.
+ */
+class ProxyFactory {
+  /**
+   * Constructor of UserWalletFactory.
+   *
+   * @param {Web3} auxiliaryWeb3 Auxiliary chain web3 object.
+   * @param {String} address Gnosis safe proxy address of a user.
+   */
+  constructor(auxiliaryWeb3, address) {
+    if (!(auxiliaryWeb3 instanceof Web3)) {
+      const err = new TypeError("Mandatory Parameter 'auxiliaryWeb3' is missing or invalid.");
+      return Promise.reject(err);
+    }
+
+    if (!Web3.utils.isAddress(address)) {
+      const err = new TypeError(`Mandatory Parameter 'address' is missing or invalid: ${address}.`);
+      return Promise.reject(err);
+    }
+
+    this.auxiliaryWeb3 = auxiliaryWeb3;
+    this.address = address;
+
+    this.contract = Contracts.getProxyFactory(this.auxiliaryWeb3, this.address);
+
+    if (!this.contract) {
+      const err = new TypeError(`Could not load ProxyFactory contract for: ${this.address}`);
+      return Promise.reject(err);
+    }
+  }
+
+  /**
+   * Deploys proxyFactory contract.
+   *
+   * @param {Web3} auxiliaryWeb3 Origin chain web3 object.
```

Comment:
```suggestion
   * @param {Web3} auxiliaryWeb3 Auxiliary chain web3 object.
```
Please fix across the methods.